### PR TITLE
fix multi running container for the same pod.Spec.Containers

### PR DESCRIPTION
**Pull request by xiaoanyunfei**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

image

>  I0412 14:30:49.476501 2665331 kuberuntime_container.go:412] ContainerStatus for 4705f32c460eba4e7327068d4c2748be1bc3e731f5641c93ffe75d82bf680834 error: rpc error: code = Unknown desc = Error: No such container: 4705f32c460eba4e7327068d4c2748be1bc3e731f5641c93ffe75d82bf680834
 
> E0412 14:33:42.738180 2665331 remote_runtime.go:222] StartContainer "4705f32c460eba4e7327068d4c2748be1bc3e731f5641c93ffe75d82bf680834" from runtime service failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded

when docker response time out, the status of container is `Unknown`, and kubelet will try to create a new container with restart+1. There may end up with two same running container with different restart count, and kubelet can never kill the older one.

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/sig node

### CODE REVIEW

**Comment by xiaoanyunfei** [comment]
triage/accepted 
/assign @dchen1107

File: pkg/kubelet/container/runtime.go [hunk 1]
```diff
@@ -346,6 +346,18 @@ func (podStatus *PodStatus) FindContainerStatusByName(containerName string) *Sta
 	return nil
 }
 
+// GetAllContainerStatusesByName return all running container status in the pod status with the given name.
+// the containers have already been sorted by CreatedAt
+func (podStatus *PodStatus) GetAllContainerStatusesByName(containerName string) []*Status {
```
Comment (ehashman) [inline pkg/kubelet/container/runtime.go:351]: This is almost a duplicate of `FindContainerStatusByName`:

https://github.com/kubernetes/kubernetes/blob/032a6fd49429f5d09814c43a243ea74f6c64cbe9/pkg/kubelet/container/runtime.go#L338-L340

We should probably be consistent in the naming.

**Review by ehashman — COMMENTED** [review]
I like the idea of this PR is correct but I don't think the approach is right. There are a lot of places `FindContainerStatusByName` is used and none are resilient to multiple statuses being returned. I think we should improve that code to be more robust and handle this everywhere, rather than special-casing this here.

**Pull request closed without merging**